### PR TITLE
docs: automated address management updates

### PR DESCRIPTION
**Pull request by rvantonder**
## Description 

Updates docs to reflect https://github.com/MystenLabs/sui/pull/18470 changes.

## Test plan 

N/A docs

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:

### CODE REVIEW

**Review by amnn — APPROVED** [review]
👍

File: docs/content/concepts/sui-move-concepts/packages/upgrade.mdx [hunk 1]
```diff
@@ -440,12 +425,13 @@ published-at = "<ORIGINAL-PACKAGE-ID>"
 sui_package = "0x0"
 ```
 
+With the new manifest and code in place, we can proceed.
```
Comment (ronny-mysten) [inline docs/content/concepts/sui-move-concepts/packages/upgrade.mdx:428]:
```suggestion
With the new manifest and code in place, you can proceed.
```

File: docs/content/concepts/sui-move-concepts/packages/upgrade.mdx [hunk 2]
```diff
@@ -377,23 +377,28 @@ The result includes an **Object changes** section with two pieces of information
...
-Add an entry for the package in the `[addresses]` section that points to the value of the on-chain ID:
+:::info
 
-```toml
-[addresses]
-sui_package = "<ORIGINAL-PACKAGE-ID>"
-```
+Beginning with the Sui `v1.29.0` release, published addresses are automatically managed in the `Move.lock` file and you do not need to take further action.
+
+If the package was published or upgraded with a Sui version prior to `v1.29.0`, you may follow [the guide for adopting automated address management](automated-address-management#adopting-automated-address-management-for-published-packages). Alternatively, refer to the `Manual Addresses` tab above for further steps.
```
Comment (ronny-mysten) [inline docs/content/concepts/sui-move-concepts/packages/upgrade.mdx:388]:
```suggestion
If the package was published or upgraded with a Sui version prior to `v1.29.0`, you can follow [the guide for adopting automated address management](automated-address-management#adopting-automated-address-management-for-published-packages). Alternatively, refer to the `Manual Addresses` tab above for further steps.
```

**Review by ronny-mysten — APPROVED** [review]
👍 👍

**Pull request merged**